app/app.go

```diff
@@ -1096,14 +1114,92 @@ func (app *App) InitChainer(ctx sdk.Context, req abci.RequestInitChain) abci.Res
 	return app.mm.InitGenesis(ctx, app.appCodec, genesisState, app.genesisImportConfig)
 }
 
-func (app *App) PrepareProposalHandler(_ sdk.Context, req *abci.RequestPrepareProposal) (*abci.ResponsePrepareProposal, error) {
+func (app *App) PrepareProposalHandler(ctx sdk.Context, req *abci.RequestPrepareProposal) (*abci.ResponsePrepareProposal, error) {
+	// Get all pending bundles for this height
+	fmt.Println("Preparing proposal for height", ctx.BlockHeight())
```

Comment:
does the sdk context have a logger? I'd avoid println's in "production" code

Reply:
it does, something like
```suggestion
	ctx.Logger().Debug("Preparing proposal", "height", ctx.BlockHeight())
```